pennylane_lightning_gpu/src/tests/Test_StateVectorCudaManaged_NonParam.cpp

```diff
@@ -830,3 +830,98 @@ TEMPLATE_TEST_CASE("StateVectorCudaManaged::applyCSWAP",
+            else
+                matrix[i] = std::complex<TestType>(0, 0);
+        }
+
+        auto results = svdat.cuda_sv.expval(wires, matrix);
+
+        cp_t expected(1, 0);
+        CHECK(expected.real() == Approx(results.x).epsilon(0.001));
+        CHECK(expected.imag() == Approx(results.y).epsilon(0.001));
+    }
+
+    SECTION("GetExpectionRandomMatrix") {
+        using cp_t = std::complex<TestType>;
+        const std::size_t num_qubits = 3;
+        std::vector<cp_t> init_state{{0.0, 0.0}, {0.0, 0.1}, {0.1, 0.1},
+                                     {0.1, 0.2}, {0.2, 0.2}, {0.3, 0.3},
+                                     {0.3, 0.4}, {0.4, 0.5}};
+        SVDataGPU<TestType> svdat{num_qubits};
+        svdat.cuda_sv.CopyHostDataToGpu(init_state.data(), init_state.size());
+        std::vector<size_t> wires{0, 1, 2};
+        std::vector<cp_t> matrix{
+            {0.5, 0.0},  {0.2, 0.5},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.5, 0.0},  {0.2, 0.5},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.5, 0.0},  {0.2, 0.5},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.5, 0.0},  {0.2, 0.5},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0}};
+
+        auto results = svdat.cuda_sv.expval(wires, matrix);
+
+        cp_t expected(1.263000, -1.011000);
+
+        CHECK(expected.real() == Approx(results.x).epsilon(0.001));
+        CHECK(expected.imag() == Approx(results.y).epsilon(0.001));
+    }
+}
+
+TEMPLATE_TEST_CASE("StateVectorCudaManaged::Hamiltonian_expval_cuSparse",
+                   "[StateVectorCudaManaged_Nonparam]", float, double) {
+    using cp_t = std::complex<TestType>;
+    const std::size_t num_qubits = 3;
+
+    SECTION("GetExpectionCuSparse") {
+        using cp_t = std::complex<TestType>;
+        const std::size_t num_qubits = 3;
+        std::vector<cp_t> init_state{{0.0, 0.0}, {0.0, 0.1}, {0.1, 0.1},
+                                     {0.1, 0.2}, {0.2, 0.2}, {0.3, 0.3},
+                                     {0.3, 0.4}, {0.4, 0.5}};
+        SVDataGPU<TestType> svdat{num_qubits};
+        svdat.cuda_sv.CopyHostDataToGpu(init_state.data(), init_state.size());
+        int csrOffsets[9] = {0, 2, 4, 6, 8, 10, 12, 14, 16};
+        int columns[16] = {0, 3, 1, 2, 1, 2, 0, 3, 4, 7, 5, 6, 5, 6, 4, 7};
```

Comment:
I assume we have no choice but to use `int` as the datatype here?

Reply:
Thanks for pointing out! The cusparse official doc shows that the data type of csrOffset, columns is void  and the type of nzz should be int64_t. Will update them soon